RoaringBitmap/src/main/java/org/roaringbitmap/BitmapBatchIterator.java

```diff
@@ -54,4 +54,22 @@ void wrap(BitmapContainer bitmap) {
     word = bitmap.bitmap[0];
     this.wordIndex = 0;
   }
+
+  /**
+   * Advance iterator such that next value will be greater or equal to minVal
+   *
+   * @param minVal - expected minimal value
+   */
+  public void advanceIfNeeded(char minVal) {
+    int newWordIndex = minVal >> 6;
+
+    if (wordIndex < newWordIndex) {
+      word = bitmap.bitmap[newWordIndex];
+      wordIndex = newWordIndex;
+    }
+
+    if ((64 * wordIndex) + numberOfTrailingZeros(word) < minVal) {
```

Comment:
Can you explain why you add this check here? It is likely to translate into a significant performance penalty.

Reply:
I agree, It is worrying me too. The reason is to prevent clearing bits from current word if someone after going forward, tried to advance backwards.

advanceIfNeeded(it, 84);
advanceIfNeeded(it, 30);

I have created unit tests that cover this case:
name: testAdvanceIfNeededToStalePositionDontClearCurrentBits, in ContainerBatchIteratorTest class

alternative approach that I came up is to create mask

int mask = ~(-((newWordIndex - wordIndex) >>> 31));
word &= 0xFFFFFFFFFFFFFFFFL << ((minVal & mask) & 0x3F);

If anyone has idea for more performant creation of such mask or achieving same outcome, please share It with me

Reply:
Can we catch it right before?

```Java
if(wordIndex > newWordIndex) {
  return; // nothing todo
} else  if (wordIndex < newWordIndex) {
 ... same as currently
}
```

It would still add a branch, so it might slow things down a bit, but it would just be a compare and jump instead of the more expensive routine you are currently doing.

Reply:
Yes we can, we could extend this approach further and invert check to act only when wordIndex is lesser or equal to newWordIndex. What do you think ?


    int newWordIndex = minVal >> 6;

    if (wordIndex <= newWordIndex) {
      if (wordIndex < newWordIndex) {
        word = bitmap.bitmap[newWordIndex];
        wordIndex = newWordIndex;
      }

      word &= 0xFFFFFFFFFFFFFFFFL << (minVal & 0x3F);
    }

Reply:
That looks even better.